### PR TITLE
Adds Flood Agent support package

### PR DESCRIPTION
**Pull request by ivanvanderbyl**

### CODE REVIEW

File: packages/flood-runner/src/ConcurrencyHack.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { readdirSync } from 'fs'
+import { join } from 'path'
+
+const lockdirPath = join(process.env.FLOOD_DATA_ROOT ?? '/data/flood', 'concurrency.lock')
```
Comment (lachie) [inline packages/flood-runner/src/ConcurrencyHack.ts]: hey, this needs to be a subdir of FLOOD_DATA_ROOT

File: packages/flood-runner/src/Environment.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		SEQUENCE: this.threadID * this.gridSequenceID * (this.nodeIndex + 1),
+		FLOOD_LOAD_TEST: true,
+	}
+}
+
+export function initFromEnvironment(env: ProcessEnv = process.env): Partial<GridConfig> {
+	let root: string
+	let testDataRoot: string
+
+	if (env.NODE_ENV !== 'production') {
```
Comment (lachie) [inline packages/flood-runner/src/Environment.ts]: will we have to set this from flood-agent?

File: packages/flood-runner/src/Environment.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		devDefaults(env)
+		const projectRoot = findRoot(__dirname)
+		root = path.join(projectRoot, 'tmp/data')
+		testDataRoot = path.join(projectRoot, 'tmp/data/flood/files')
+	} else {
+		root = '/data'
```
Comment (lachie) [inline packages/flood-runner/src/Environment.ts]: this isn't necessarily true anymore (outside docker)... should be set from `$FLOOD_DATA_ROOT`

File: packages/flood-runner/src/Environment.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		testDataRoot = env.TEST_DATA_DIRECTORY || '/data/flood/files'
```
Comment (lachie) [inline packages/flood-runner/src/Environment.ts]: `$FLOOD_FILES_PATH` is probably better now

File: packages/flood-runner/src/ConcurrencyHack.ts (hunk 1 above)
Comment (ivanvanderbyl) [reply packages/flood-runner/src/ConcurrencyHack.ts]: any pref of what it is? this is assumed to exist by Element

File: packages/flood-runner/src/Environment.ts (hunk 2 above)
Comment (ivanvanderbyl) [reply packages/flood-runner/src/Environment.ts]: If it's unset it's okay, I assume this is just for smoke testing/legacy tests

**Pull request merged**